crates/sui/src/keytool.rs

```diff
@@ -257,6 +281,88 @@ impl KeyToolCommand {
                     }
                 }
             }
+            KeyToolCommand::SignKMS {
+                data,
+                keyid,
+                intent
+            } => {
+                // Currently only supports secp256k1 keys
+                println!("Raw tx_bytes to execute: {}", data);
+                let intent = intent.unwrap_or_else(Intent::sui_transaction);
+                println!("Intent: {:?}", intent);
+                let msg: TransactionData =
+                    bcs::from_bytes(&Base64::decode(&data).map_err(|e| {
+                        anyhow!("Cannot deserialize data as TransactionData {:?}", e)
+                    })?)?;
+                let intent_msg = IntentMessage::new(intent, msg);
+                println!(
+                    "Raw intent message: {:?}",
+                    Base64::encode(bcs::to_bytes(&intent_msg)?)
+                );
+                let mut hasher = DefaultHash::default();
+                hasher.update(bcs::to_bytes(&intent_msg)?);
+                let digest = hasher.finalize().digest;
+                println!("Digest to sign: {:?}", Base64::encode(digest));
+               
+                // Set up the KMS client, expect to use env vars or whatever the AWS
+                // SDK prefers.
+                // Length of Pubkey - 33, sig - 64
+                
+                let region = Region::default();
+                let kms = KmsClient::new(region);
+
+                // Get Pub key from keyid. This is in DER Format from AWS
+                let req = GetPublicKeyRequest { grant_tokens: None, key_id: keyid.to_string() };
+                let pubkey_resp = kms.get_public_key(req).await.unwrap();
```

Comment:
avoid unwrap in general since it leads to undesirable panics, the function returns anyhow so you can always map to anyhow! and return to caller.

Reply:
updated